### ISSUE TITLE
LC0003 / LC0005 False positives on DotNet-related symbols in arrays, WithEvents triggers, and usercontrol add-ins

### ISSUE THREAD

**Issue by mikkel-bcw** [StefanMaron/BusinessCentral.LinterCop#1204]
I’m seeing **false positive warnings** from the linter on DotNet-related types/symbols in a few scenarios. The code compiles and runs as expected, but the linter reports warnings like **LC0003** and **LC0005** on valid symbols.

### Example 1 — DotNet type inside array

<img width="859" height="84" alt="Image" src="https://github.com/user-attachments/assets/13db5a89-4425-467e-9cbd-68da518198d3" />

Linter reports **LC0003** and **LC0005** on `XmlElement`.

```al
var
    XmlElements: array[10] of DotNet XmlElement;
```

---

### Example 2 — WithEvents trigger method name flagged

<img width="790" height="79" alt="Image" src="https://github.com/user-attachments/assets/190d338e-93b6-477e-bb2e-86602b7043a3" />

Linter reports **LC0005** on the event trigger `NewMessage`.

```al
var
    [WithEvents]
    MessagingServer: DotNet MessagingServer;

trigger MessagingServer::NewMessage(messagingResult: DotNet MessagingResult)
begin
    // ...
end;
```

---

### Example 3 — usercontrol AddIn type flagged

Linter reports **LC0005** on `BusinessChartAddIn` used in `usercontrol`.

```al
dotnet
{
    assembly("Microsoft.Dynamics.Nav.Client.BusinessChart")
    {
        type("Microsoft.Dynamics.Nav.Client.BusinessChart.BusinessChartAddIn"; BusinessChartAddIn)
        {
            IsControlAddIn = true;
        }
    }
}
page 56789 MyPage
{
    PageType = CardPart;

    layout
    {
        area(Content)
        {
            usercontrol(BusinessChart; BusinessChartAddIn)
            {
                // ...
            }
        }
    }
}
```
